go/arrow/flight/flightsql/driver/driver.go

```diff
@@ -37,34 +38,50 @@ import (
 )
 
 type Rows struct {
-	schema        *arrow.Schema
-	records       []arrow.Record
-	currentRecord int
-	currentRow    int
+	schema           *arrow.Schema
+	recordChan       chan arrow.Record
+	currentRecord    arrow.Record
+	currentRecordMux sync.Mutex
+	currentRow       uint64
+	initializedChan  chan bool
+	streamError      error
+}
+
+func newRows() *Rows {
+	return &Rows{
+		recordChan:      make(chan arrow.Record, 1),
```

Comment:
This buffer size should probably be configurable, depending on the server in use and how the consumer is operating, getting better performance and backpressure may require manipulating the size of the buffer.

Reply:
thanks, @zeroshade .
I removed the magic number, but kept the buffer size=one.
Do you have suggestions for implementing a dynamic/configurable buffer size?
That could, anyway, happen on a new PR. WDYT?

Reply:
I'd be fine with making it configurable in a follow-up PR. sure. 

As for a suggestion for implementing the configuable buffer size, there are two common patterns:

* In the `Open` and `Connector` handling the uri, you can use a delimiter and allow passing various options in similar to how an ODBC DSN works. During `Open`  you can then parse out the options and store them in the driver instance, then have this function take in the buffer size as a parameter which can be passed in.
* You could provide context functions like `WithBufferSize` and then pull the value out of the context at query time when you construct the Rows object, defaulting to 1 if the user didn't stick it into the context.